Ok, sorry for the delay buy I had some problem setting up a development env for ansible 1.9

`tls_hostname` works as expected. Without it I can't authenticate the server.
Environment variables work as expected, if explicit parameters are not specified. (I got confused by this initially =D)

With `use_tls=encrypt` (and nothing else) I get this:

```
msg: SSLError(SSLError(SSLError(1, '_ssl.c:507: error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed'),),)
```

I think this is beacuse we must set tls_verify to False. For instance:

``` diff
diff --git a/cloud/docker/docker.py b/cloud/docker/docker.py
index 1014fc1..e945b2c 100644
--- a/cloud/docker/docker.py
+++ b/cloud/docker/docker.py
@@ -609,6 +611,7 @@ class DockerManager(object):
             if params or use_tls == 'encrypt':
                 # See https://github.com/docker/docker-py/blob/d39da11/docker/utils/utils.py#L279-L296
                 docker_url = docker_url.replace('tcp://', 'https://')
+                params['verify'] = use_tls != 'encrypt'
                 tls_config = docker.tls.TLSConfig(**params)

         self.client = docker.Client(base_url=docker_url,
```

Last, using just these two: `use_tls=verify tls_hostname=boot2docker` it exits for the wrong reason:

```
TASK: [Start MySQL container] ************************************************* 
failed: [localhost] => {"failed": true, "parsed": false}
Traceback (most recent call last):
  File "/Users/davide/.ansible/tmp/ansible-tmp-1426185465.11-224963353501429/docker", line 3092, in <module>
    main()
  File "/Users/davide/.ansible/tmp/ansible-tmp-1426185465.11-224963353501429/docker", line 1441, in main
    manager = DockerManager(module)
  File "/Users/davide/.ansible/tmp/ansible-tmp-1426185465.11-224963353501429/docker", line 607, in __init__
    params['assert_hostname'] = tls_hostname
UnboundLocalError: local variable 'tls_hostname' referenced before assignment


FATAL: all hosts have already failed -- aborting
```

This fixes the issue:

``` diff
diff --git a/cloud/docker/docker.py b/cloud/docker/docker.py
index 1014fc1..0a827c9 100644
--- a/cloud/docker/docker.py
+++ b/cloud/docker/docker.py
@@ -576,6 +576,8 @@ class DockerManager(object):
                     tls_hostname = parsed_url
             if not tls_hostname:
                 tls_hostname = True
+        else:
+            tls_hostname = None

         # use_tls can be one of four values:
         # no: Do not use tls
```

With this patch the error is:

```
msg: SSLError(SSLError(SSLError(1, '_ssl.c:507: error:14090086:SSL routines:SSL3_GET_SERVER_CERTIFICATE:certificate verify failed'),),)
```

which makes more sense. But you may want to warn the user even before as `use_tls=verify` should require CA at least.

--- a/cloud/docker/docker.py
+++ b/cloud/docker/docker.py
@@ -567,19 +567,18 @@ class DockerManager(object):
         if not tls_ca_cert and env_cert_path:
             tls_ca_cert = os.path.join(env_cert_path, 'ca.pem')
 
-        if tls_ca_cert:
-            tls_hostname = module.params.get('tls_hostname')
-            if tls_hostname is None:
-                if env_docker_hostname:
-                    tls_hostname = env_docker_hostname
+        tls_hostname = module.params.get('tls_hostname')
+        if tls_hostname is None:
+            if env_docker_hostname:
+                tls_hostname = env_docker_hostname
+            else:
+                parsed_url = urlparse(docker_url)
+                if ':' in parsed_url.netloc:
+                    tls_hostname = parsed_url.netloc[:parsed_url.netloc.rindex(':')]
                 else:
-                    parsed_url = urlparse(docker_url)
-                    if ':' in parsed_url.netloc:
-                        tls_hostname = parsed_url.netloc[:parsed_url.netloc.rindex(':')]
-                    else:
-                        tls_hostname = parsed_url
-            if not tls_hostname:
-                tls_hostname = True
+                    tls_hostname = parsed_url
+        if not tls_hostname:
+            tls_hostname = True
 
         # use_tls can be one of four values:
         # no: Do not use tls
@@ -610,8 +609,10 @@ class DockerManager(object):
                 else:
                     params['verify'] = True
                     params['assert_hostname'] = tls_hostname
+            elif use_tls == 'encrpyt':
+                params['verify'] = False
 
-            if params or use_tls == 'encrypt':
+            if params:
                 # See https://github.com/docker/docker-py/blob/d39da11/docker/utils/utils.py#L279-L296
                 docker_url = docker_url.replace('tcp://', 'https://')
                 tls_config = docker.tls.TLSConfig(**params)

